### DISCUSSION TITLE
深度定制开发问题

### DISCUSSION THREAD

**Discussion by hehos** [dream-num/univer#1733] [Q&A | 常见问题]
univer团队成员们，大家好！
首先感谢你们对开源做出的奉献。
univer最近看文档做了详细的了解，现在在读源码深入了解项目的底层原理及架构体系。对该项目的插件架构体系，模型分层等等架构设计觉得很好。于是有了一个想法：
因为公司研发的需要。公司计划做一些像报表、结构化文档类似的项目。在做技术选型上，我决定借鉴该项目来做深度的定制开发，实现前面提到的公司项目。
同时，在深度了解该项目后，我也非常希望在将来参与到该项目的开发和维护，为该项目和开源社区奉献我微薄之力。
对于上面的深度定制化开发的想法，不知道univer的开发者们有什么好的建议或者指点呢？

**Comment by jikkai** [comment]
感谢你对 Univer 的关注。

Univer 基于 Apache 2.0 开源，允许用户自由地使用、修改和分发，无论是用于商业还是非商业用途。 因此你可以随意对本仓库 fork 并进行改动。我们也非常欢迎开发者以任何形式参与到共建 Univer 的工作中。

如果你在开发中遇到了 Univer 中存在的 bugs 或者有好的想法，也欢迎提 [issues](https://github.com/dream-num/univer/issues) 告知我们。

**Reply by hehos** [reply]
Univer 适合深度定制开发报表、结构化文档类似的项目吗？想听听你们的看法？

**Reply by jikkai** [reply]
这仅取决于你对自身业务场景的判断。

Univer 作为一个可集成的开源办公软件套装，并不会向社区提供诸如使用范围、适用场景之类的建议。
